lua/entities/gmod_wire_expression2/core/selfaware.lua

```diff
@@ -91,29 +91,75 @@ end)
 /******************************************************************************/
 -- Name functions
 
--- Set the name of the E2 itself
-e2function void setName( string name )
-	local e = self.entity
-	if (e.name == name) then return end
-	if (name == "generic" or name == "") then
+local CurTime = CurTime
+
+local antiSpamLookup = {}
+local function NameSpamCheck(ent) -- Returns true if it is being spammed; otherwise false
+	if antiSpamLookup[ent] and antiSpamLookup[ent] > CurTime() then
+		return true
+	end
+	antiSpamLookup[ent] = CurTime() + 0.1 -- Adjust the delay here, it is applied to setter (setName/setComponentName) functions
+	return false
+end
+
+local function SetE2Name(ent, name)
+	if NameSpamCheck(ent) or ent.name == name then return end
+	if name == "generic" or name == "" then -- NOTE: This could be simplified if we exclude the "generic" check
 		name = "generic"
-		e.WireDebugName = "Expression 2"
+		ent.WireDebugName = "Expression 2"
 	else
-		e.WireDebugName = "E2 - " .. name
+		ent.WireDebugName = "E2 - " .. name
+	end
+	ent.name = name
+	ent:SetNWString("name", name)
+	ent:SetOverlayText(name)
+end
+
+__e2setcost(20)
+
+local IsValid = IsValid
+
+-- Set the name of the E2 itself
+e2function void setName(string name)
+	local e = self.entity
+	if IsValid(e) then
+		SetE2Name(e, name)
+	end
+end
+
+-- Set the name of another E2
+e2function void entity:setName(string name)
+	if IsValid(this) and this:GetClass() == "gmod_wire_expression2" and E2Lib.getOwner(this) == self.player then
+		SetE2Name(this, name)
 	end
-	e.name = name
-	e:SetNWString( "name", e.name )
-	e:SetOverlayText(name)
 end
 
--- Get the name of another E2
+local string_find = string.find
+local duplicator_StoreEntityModifier = duplicator.StoreEntityModifier
+
+-- Set the component name of Wire-compatible entity
+e2function void entity:setComponentName(string componentName)
+	if not IsValid(this) or NameSpamCheck(this) or E2Lib.getOwner(this) ~= self.player or this.wireName == componentName or string_find(componentName, "[\n\r\"]") ~= nil then return end
+	this.wireName = componentName
+	this:SetNWString("WireName", componentName)
+	duplicator_StoreEntityModifier(this, "WireName", { name = componentName })
```

Comment:
@TomyLobo Ok, on Steam, to my understanding, you have told me to move those two lines into WireLib. However, there is a question you haven't answered me yet:
So if I move it into WireLib, as you requested, will I also have to change Namer/Indicator stool and other places too to use WireLib function?